### PR TITLE
Use SnoopPrecompile and Fix precompile breakage on nightly

### PR DESCRIPTION
**Pull request by sefffal**
Precompile was broken on nightly since the precompile signature for keyword arguments is now different.
This uses SnoopPrecompile to generate precompile statements automatically which should fix the issue and as a side effect, improve TTFX for some basic common operations on 2D matrices.

I'd love a way to precompile the png display methods but not sure how to do this without actually showing something during precompile.
Maybe `save(tempname()*".png", img)` would help.

### CODE REVIEW

File: src/precompile.jl [hunk 1]
```diff
@@ -14,31 +13,28 @@ for T in [Float32, Float64, Int, Int8, UInt8, N0f8]
...
+end
```
Comment (giordano) [inline src/precompile.jl:40]: Missing new line
```suggestion
end
```
Like in https://github.com/JuliaAstro/AstroImages.jl/commit/1f7b2c66df36042c9d93763fc6bac12a9ed1f6e7#diff-5aa37a40d0f0d9f8a355bf14071e63cd81627fe63730c12992c1d63e2977afeaR45 BTW

**Comment by sefffal** [comment]
This PR does not actually improve any timings so I guess the previous calls were all inferred.
I'm still in favour of it though since it should make it easy to precompile more calls in future.

**Comment by giordano** [comment]
Out of curiosity, how are you measuring TTFX? Do you have a reproducible script?

**Comment by sefffal** [comment]
Sure. With and without these changes, I started julia, ran `] precompile AstroImages` and then:
```
julia> @time @time_imports using AstroImages
```
followed by
```
julia> @time imview(randn(10,10));
julia> @time imview(AstroImage(randn(10,10)));
```

I figured that seeing the first image displayed would be the most noticeable lag.
These functions were in the precompile script before as well so I guess that means it was doing a decent job.
If there was a way to include the actual display machinery in the precompile maybe we would see a bigger difference?

**Comment by giordano** [comment]
At the moment I get
```julia
julia> @time using AstroImages
  2.255676 seconds (4.17 M allocations: 310.222 MiB, 6.09% gc time, 1.38% compilation time: 7% of which was recompilation)

julia> @time imview(randn(10,10));
  0.429312 seconds (554.36 k allocations: 34.775 MiB, 99.97% compilation time)

julia> @time imview(AstroImage(randn(10,10)));
  0.013798 seconds (4.83 k allocations: 353.492 KiB, 98.80% compilation time)
```
which isn't too bad.  Loading the package is the most annoying part, but it's mostly due to the Image* packages.

**Comment by sefffal** [comment]
Yes I guess it is good enough for now.

In terms of using-time, the biggest dependencies are:
```
    856.7 ms  ImageCore
    502.7 ms  PlotUtils
    381.2 ms  Colors
    332.8 ms  FileIO 6.32% compilation time (17% recompilation)
    286.9 ms  ColorVectorSpace 2.73% compilation time
    250.7 ms  DimensionalData
```
I don't think there's anything here that can really be avoided.

**Comment by KristofferC** [comment]
New release with this would be nice :)

**Pull request merged**